### ISSUE TITLE
Regression in first item in Switcher/Fast Switcher for 0.8.0

### ISSUE THREAD

**Issue by vaeth** [WayfireWM/wayfire#2124]
Hi,

in wayfire-0.8.0, there was a change in the behavior of Switcher and Fast Switcher:

When e.g. switcher is configured for Alt-Tab and you have many windows open, you can no longer rotate through the whole collection of windows by successively pressing Alt-Tab (when you leave both keys). Of course, you can still rotate through the whole collection when you keep the Alt-key pressed, but just using "quickly" Alt-Tab to see the "next" window is no longer possible: Essentially, you are now only be able to toggle between two windows unless you press tab several time with keeping the Alt-key pressed.

I am not sure whether this change is intentional or not. Some people might like it, but I find it rather disturbing for all of my workflows, since keeping the Alt-key pressed was never necessary previously to get to any window I wanted.
Moreover, the functionality of the "next view" is essentially melted to the functionality to the "previous view" key, since both allow you to toggle only between the same two windows (unless you keep the alt-key pressed for a long time which is rather inconvenient).

I guess that the change is due to some deeper internal change of wayfire (about how the window list is organized) so that after switching to a new window for some reason the "next" window in the window list becomes reorganized as the previously active window. Anyway, it would be nice if there would be an option to revert the previous behavior, either by making the plugins somehow work around the problem of the modified internal organisation or if the internal organization (when activating to a new window) itself could be modified with some option.

**Comment by vaeth** [comment]
I just realized that it might be that the old behavior is re-obtained with https://github.com/WayfireWM/wayfire/pull/2119, but I did not yet check anything newer than 0.8.0.

**Comment by ammen99** [comment]
I believe you must be mistaken - I just tested Wayfire 0.7.2 and the behavior of fast-switcher hasn't changed, so this is specific to the switcher plugin.

In any case, the behavior of switcher was changed in https://github.com/WayfireWM/wayfire/pull/1812 . The author's PR had proposed to make this change optional, but at that point it seemed there was nobody who wanted to use the old behavior so we made the new behavior mandatory. Feel free to send a PR to introduce an option to control this and to use the old behavior instead of the new.

**Comment by vaeth** [comment]
> I believe you must be mistaken - I just tested Wayfire 0.7.2 and the behavior of fast-switcher hasn't changed

Probably you are right: I had always used switcher and tried fast-switcher only in wayfire 0.8.0 when I realized that switcher no longer worked as expected - so I expected they changed behavior simultaneously without verifying it now once more before posting.

Thank you for pointing out  the relevant PR. When I find the time, I will create a PR to make it optional (or patch it manually if I do not find the time...)

**Comment by vaeth** [comment]
I removed "Fast Switcher" from the title to make the report not misleading.

Moreover, after reading that the change was intentional, I experimented and found a workaround for "Fast Switcher":

For "Fast Switcher", I could just define "Alt-Tab" for the **previous** view on "Super-Tab" for the **next** view and then have essentially the previous behavior of "Switcher". For "Switcher" this workaround does not work: The previous view and the next view both toggle just between the same two windows. Maybe I report this as a separate issue.
